### PR TITLE
initial chunk generation ready to go

### PR DESCRIPTION
**Pull request by til61**
This commit allows generation of one 80 * 48 tiles background without scrolling
Warning: This will not compile until wfc.rs is in place.
The text to generate rules from is included in this commit at assets/backgrounds/input.txt

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,6 +7,7 @@ edition = "2021"
 
 [dependencies]
 bevy = { version = "0.8.1", features = ["dynamic"] }
+rand = "0.8.4"
```
Comment (Elsklivet) [inline Cargo.toml:10]: Is Farnan okay with us using the rand crate?

File: src/backgrounds.rs [hunk 2]
```diff
@@ -1,11 +1,74 @@
...
 
 #[derive(Component)]
 pub(crate) struct Background;
 
+pub(crate) fn init_background(mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut texture_atlases: ResMut<Assets<TextureAtlas>>
+) {
+
+    let starting_chuck = wfc();
```
Comment (Elsklivet) [inline src/backgrounds.rs:22]: Think you meant to name this `starting_chunk`

```suggestion
    let starting_chunk = wfc();
```

File: src/backgrounds.rs [hunk 3]
```diff
@@ -1,11 +1,74 @@
...
+                    translation: t,
+                    ..default()
+                },
+                sprite: TextureAtlasSprite {
+                    index: tile,
+                    ..default()
+                },
+                ..default()
+            })
+            .insert(Background);
```
Comment (Elsklivet) [inline src/backgrounds.rs:58]: Perhaps we should come up with a different component if we're gonna insert them all with the background component, such as `Tile`.

File: src/player.rs [hunk 4]
```diff
@@ -1,5 +1,5 @@
 use bevy::{prelude::*};
-use crate::backgrounds::{Background, TILE_SIZE, LEVEL_LENGTH, LEVEL_HEIGHT};
```
Comment (Elsklivet) [inline src/player.rs:2]: Wait wait wait, why were these removed?

The PR to change player movement was not correctly reviewed, with this code the player can move well past world boundaries without being stopped.

Shouldn't we put these back and put a check in `move_player`?

**Review by Elsklivet — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (til61) [reply Cargo.toml:10]: Yes.

File: src/backgrounds.rs (hunk 3 above)
Comment (til61) [reply src/backgrounds.rs:58]: Why, what else is background being used?

Comment (Elsklivet) [reply src/backgrounds.rs:58]: It's just a non-descriptive component name versus Tile. If each individual tile will be inserted with the component, each tile isn't a whole "background".

Comment (til61) [reply src/backgrounds.rs:58]: the background is originally empty, I insert tiles in it, so it becomes the full background.

Comment (Elsklivet) [reply src/backgrounds.rs:58]: Each individual tile is not a background. It's up to you but I just think it's not as descriptive of a name as Tile. Plus we could add Tile specific variables to a Tile struct and it's just clearer style wise what it refers to.
